cluster-operator/src/main/java/io/strimzi/operator/cluster/model/CruiseControl.java

```diff
@@ -419,42 +411,86 @@ protected List<EnvVar> getEnvVars() {
+                throw new RuntimeException(format("Secret %s is invalid", oldSecret.getMetadata().getName()));
+            } else if (!Util.decodeFromBase64(authFile).contains(API_ADMIN_NAME) || !Util.decodeFromBase64(authFile).contains(API_USER_NAME)) {
+                throw new RuntimeException(format("Secret %s has invalid authentication file", oldSecret.getMetadata().getName()));
+            } else {
+                return data;
+            }
+        } else {
+            String apiAdminPassword = passwordGenerator.generate();
+            String apiUserPassword = passwordGenerator.generate();
+
+            /*
+             * Create Cruise Control API auth credentials file following Jetty's
+             *  HashLoginService's file format: username: password [,rolename ...]
+             */
+            String authCredentialsFile =
+                API_ADMIN_NAME + ": " + apiAdminPassword + "," + API_ADMIN_ROLE + "\n" + 
+                    API_USER_NAME + ": " + apiUserPassword + "," + API_USER_ROLE + "\n";
+
+            if (adminUser != null) {
+                authCredentialsFile += adminUser.username() + ": " + adminUser.password() + "," + API_ADMIN_ROLE + "\n";
+            }
+
+            return Map.of(
+                API_ADMIN_PASSWORD_KEY, Util.encodeToBase64(apiAdminPassword),
+                API_USER_PASSWORD_KEY, Util.encodeToBase64(apiUserPassword),
+                API_AUTH_FILE_KEY, Util.encodeToBase64(authCredentialsFile)
+            );
+        }
+    }
+
+    /**
+     * Cruise Control user credentials.
+     * 
+     * @param username Username.
+     * @param password Password.
      */
-    public static Map<String, String> generateCruiseControlApiCredentials(PasswordGenerator passwordGenerator) {
-        String apiAdminPassword = passwordGenerator.generate();
-        String apiUserPassword = passwordGenerator.generate();
-
-        /*
-         * Create Cruise Control API auth credentials file following Jetty's
-         *  HashLoginService's file format: username: password [,rolename ...]
-         */
-        String authCredentialsFile =
-                API_ADMIN_NAME + ": " + apiAdminPassword + "," + API_ADMIN_ROLE + "\n" +
-                API_USER_NAME + ": " + apiUserPassword + "," + API_USER_ROLE + "\n";
-
-        Map<String, String> data = new HashMap<>(3);
-        data.put(API_ADMIN_PASSWORD_KEY, Util.encodeToBase64(apiAdminPassword));
-        data.put(API_USER_PASSWORD_KEY, Util.encodeToBase64(apiUserPassword));
-        data.put(API_AUTH_FILE_KEY, Util.encodeToBase64(authCredentialsFile));
-
-        return data;
+    public record CruiseControlUser(String username, String password) {
+        @Override
+        public String toString() {
+            String mask = "********";
+            return "CruiseControlUser{" +
+                "username='" + username + '\'' +
+                ", password='" + mask + '\'' +
```

Comment:
A stupid question ... can't you just do

```suggestion
                "username='" + username + "'" +
                ", password='" + mask + "'" +
```

Reply:
This is how Idea generates toString by default. In both cases, they are compiled to this:

```java
public String toString() {
    String mask = "********";
    return "CruiseControlUser{username='" + this.username + "', password='" + mask + "'}";
}
```